server/routes/messages.js

```diff
@@ -1,11 +1,47 @@
 const messagesRouter = require('express').Router();
 const {Chat, findByRecipientsOrSave} = require('../models/chat');
 const Message = require('../models/message');
+const User = require('../models/user');
 const mustBeAuthenticated = require("../middleware");
 
 //Middleware to ensure authentication
 messagesRouter.use(mustBeAuthenticated);
 
+//Route to create a new group (POST /groups)
+messagesRouter.post('/groups', async (req, res) => {
+    try {
+      //Extract necessary data from request body
+      const { groupName, userIds } = req.body;
+
+      if (!groupName || !userIds) {
+        return res
+          .status(400)
+          .send({ msg: "Invalid request. Missing required data" });
+      }
+        
+    const users = await User.find({ _id: { $in: userIds } });
+
+      //Create a new group
+      const chat = new Chat({ type: "group", participants: userIds, name: groupName });
+
+      //Save the group in the database
+        await chat.save();
```

Comment:
Check out https://github.com/DanielC34/convocafe/blob/main/server/routes/messages.js#L94. 
that's how you populate users